src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Palaso.Reporting;
+using Palaso.UI.WindowsForms.ImageToolbox;
+
+namespace Bloom.ImageProcessing
+{
+	/// <summary>
+	/// Gecko struggles with hi-res images intented for printing. Gecko chews up memory, makes for slow drawing,
+	/// or even gives up on displaying the image altogether (worse on slow machines).
+	/// This cache takes requests for images and returns lo-res versions of them.
+	/// </summary>
+	public class RuntimeImageProcessor : IDisposable
+	{
+		private readonly BookRenamedEvent _bookRenamedEvent;
+		public int TargetDimension = 500;
+		private Dictionary<string, string> _originalPathToProcessedVersionPath;
+		private string _cacheFolder;
+
+		private ImageAttributes _transparentImageAttributes;
```

Comment:
This name didn't work well for me. Suggested something that would control whether to make the images transparent. How about _convertWhiteToTransparent or just _whiteToTransparent?